bodhi/server/util.py

```diff
@@ -296,12 +296,53 @@ def markup(context, text):
         basestring: HTML representation of the markdown text
     """
 
+    # determine the bleach version installed.
+    # this is the same approach that Pagure uses to determine the bleach version
+    # https://pagure.io/pagure/pull-request/2269#request_diff
+    bleach_v = bleach.__version__.split('.')
+    for idx, val in enumerate(bleach_v):
+        try:
+            val = int(val)
+        except ValueError:  # pragma: no cover
+            pass
+        bleach_v[idx] = val
+
+    # the only difference in the bleach API that we use between v1 and v2 is
+    # the formatting of the attributes parameter. Bleach 1 only allowed you
+    # to specify attributes to be whitelisted for all whitelisted tags.
+    # Bleach 2 requires you to specify the list of attributes whitelisted for
+    # specific tags.
+    if tuple(bleach_v) >= (2, 0, 0):  # pragma: no cover
```

Comment:
OK I've got the mock working and I've got a test that reveals a slight problem - some packages (bleach included) don't always have three values in their versions. For example, the current release of bleach is 2.0, not 2.0.0. The above code actually considers bleach 2.0 to be the old API rather than the new one.

Since we only need to make a decision based on the first value in the version, I propose that we just use the first digit and not worry about converting the version to a tuple.

I'm going to write a patch that does the things I'm proposing for you and I'll just attach it on this pull request (I'm trying to build a new Bodhi 2.9.0 beta today so I've got the time and motivation.) Then you can review my changes and if you agree you can integrate them and merge. Sound good?